### DISCUSSION TITLE
Missing alignment handling for ods

### DISCUSSION THREAD

**Discussion by knulo** [PHPOffice/PhpSpreadsheet#1795] [Ideas]
Hi,
after some investigations to align cells horizontally centered I found out, that no alignment handling is implemented in
`\vendor\phpoffice\phpspreadsheet\src\PhpSpreadsheet\Writer\Ods\Content.php`

To handle alignment I added some lines after line number 285:
`// Align`
`$hAlign = $style->getAlignment()->getHorizontal();`
`$vAlign = $style->getAlignment()->getVertical();`
`if (!empty($hAlign) || !empty($vAlign)) {`
`    $writer->startElement('style:paragraph-properties');`
`    if (!empty($hAlign))`
`          $writer->writeAttribute('fo:text-align', $hAlign);`
`    if (!empty($vAlign))`
`          $writer->writeAttribute('fo:vertical-align', $vAlign);`
`    $writer->endElement();`
`}`

Seems to work well, so I would like to see this included in the code....  ;-)

Regards

**Comment by MarkBaker** [comment]
Not quite as straightforward:

Vertical Alignment is the `style:vertical-align` attribute of `style:table-cell-properties`, and has values of `top`, `middle` and `bottom` rather than `top`, `center` and `bottom`.

Horizontal Alignment is the `fo:text-align` attribute of `style:paragraph-properties`, and uses the values of `start`, `center` and `end` rather than `left`, `center` and `right`.

**Reply by knulo** [reply]
@MarkBaker Sorry for late reply - I got no notification about your comment.
Thank you for the clarification, I will correct my code appropriately.
**And** I should read the specification of ODS syntax...   :smirk:

**Comment by knulo** [comment]
@MarkBaker
For horizontal alignment it **_is_** straightforward (refer to [OASIS open Document Format](https://docs.oasis-open.org/office/v1.2/OpenDocument-v1.2-part1.html), 20.216.1). The values of `start` end `end` are for RTL languages, I think.
And my code snippet given earlier works fine for horizontal alignment, now (version 1.16.0)  after line 356 of original file:
```
$hAlign = $style->getAlignment()->getHorizontal();
if (!empty($hAlign)) {
    $writer->startElement('style:paragraph-properties');
    $writer->writeAttribute('fo:text-align', $hAlign);
    $writer->endElement();
}
```
With `start` instead of `left` it ends up in a document that have to (but could) repaired by LibreOffice. :confused:

For vertical alignment I placed after line 336 of original file (not tested yet):
```
$vAlign = $style->getAlignment()->getVertical();
if (!empty($vAlign)) {
    $writer->writeAttribute('style:vertical-align', $vAlign);
}
```

OFFTOPIC: What can I do to receive notifications about this discussion? GitHub stated: _You’re receiving notifications because you authored the thread._ but I didn't receive anything. Should I place a beta feedback?
